doc/source/operations/secret-rotation.rst

```diff
@@ -0,0 +1,523 @@
+
+       .. code:: bash
+
+          mysql -uroot -p
+
+    4. Check the current state of the ``root`` user
+
+       .. code:: bash
+
+          SELECT Host,User,Password FROM mysql.user WHERE User='root';
+
+    5. Update the password for the ``root`` user
+
+       .. code:: bash
+
+          SET PASSWORD FOR 'root'@'%' = PASSWORD('newpassword');
+
+    6. Check that the password hash has changed in the user list
+
+       .. code:: bash
+
+          SELECT Host,User,Password FROM mysql.user WHERE User='root';
+
+    7. If there are any remaining root users with the old password e.g.
+       ``root@localhost``, change the password for them too
+
+    8. Update ``database_password`` in ``passwords.yml`` with your new
+       password
+
+
+.. _nova-change:
+
+14. Update the Nova Database password
+      .. warning::
+
+         From this point onward, service may be disrupted
+
+
+    #. Create a new ``nova_database_password`` and store it in
+       ``passwords.yml``
+
+       .. code:: bash
+
+          pwgen -s 40 1
+
+    #. Exec into the ``nova_conductor`` container
+
+       .. code:: bash
+
+          sudo docker exec -it nova_conductor bash
+
+    #. List the cells
+
+       .. code:: bash
+
+          nova-manage cell_v2 list_cells --verbose
+
+    #. Find the entry for ``cell0``, copy the Database Connection value,
+       replace the password in the string with the new value, and update it
+       with the following command:
+
+       .. code:: bash
+
+          nova-manage cell_v2 update_cell --cell_uuid 00000000-0000-0000-0000-000000000000 --database_connection "CONNECTION WITH NEW PASSWORD HERE" --transport-url "none:///"
+
+       (If the ``cell_uuid`` for cell0 is not
+       ``00000000-0000-0000-0000-000000000000``, change the above command
+       accordingly)
+
+
+15.  Re-encrypt your ``passwords.yml`` file
+
+
+.. _k-a-change:
+
+16. Delete the service users in Keystone. The exact users will depend on the
+    deployment. Multinode example:
+
+      .. note::
+
+         Alternatively, cherry-pick
+         `this patch <https://review.opendev.org/c/openstack/kolla-ansible/+/903178>`__
+
+
+    .. code:: bash
+
+      openstack user delete glance cinder placement nova neutron heat magnum magnum_trustee_domain_admin barbican designate
+
+17. Stop services using RabbitMQ
+
+    .. code:: bash
+
+       kayobe playbook run $KAYOBE_CONFIG_PATH/ansible/stop-openstack-services.yml
+
+18. Nuke RabbitMQ
+
+    .. code:: bash
+
+       kayobe overcloud host command run -l controllers --become --command "docker stop rabbitmq && docker rm rabbitmq && docker volume rm rabbitmq"
+
+19. Reconfigure Overcloud services to apply changes
+
+
+      .. warning::
+
+         VMs should continue running, but connections to them will briefly be
+         disrupted when Neutron is redeployed
+
+   .. code:: bash
+
+      kayobe overcloud service deploy
+
+
+20. Flush the Memcached data on all controllers (any old data will now be
```

Comment:
Do this after stopping services, before service deploy?

Reply:
Done